framework/events/src/main/java/org/apache/cloudstack/framework/events/EventDistributorImpl.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cloudstack.framework.events;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.cloud.utils.component.ManagerBase;
+
+public class EventDistributorImpl extends ManagerBase implements EventDistributor {
+
+    List<EventBus> eventBuses;
+
+    public void setEventBuses(List<EventBus> eventBuses) {
+        this.eventBuses = eventBuses;
+    }
+
+    @PostConstruct
+    public void init() {
+        if (logger.isTraceEnabled()) {
+            logger.trace(String.format("Found %d event buses : %s", eventBuses.size(),
+                    StringUtils.join(eventBuses.stream().map(x->x.getClass().getName()).toArray())));
+        }
```

Comment:
```suggestion
        logger.trace("Found {} event buses : {}", eventBuses.size(),
                    () -> StringUtils.join(eventBuses.stream().map(x->x.getClass().getName()).toArray()));
```

might need testing, but it does seem simpler